### PR TITLE
Teach CI to validate Web resources

### PR DESCRIPTION
**Pull request by lucafavatellawork**
TODO:
- [x] Consider not to allow HTML/CSS failure, by fixing errors.
- [x] Consider whether to restore appearance of form at `/account/account.html`.
- [x] Rebase on master once #105 merged.

### CODE REVIEW

**Comment by lucafavatellawork** [comment]
Summary of warnings.

```
'"/sse_cfg/priv/docroot/render/render.css":2.18-3.1: error: CSS: The @charset rule may only occur at the start of the style sheet. Please check that there are no spaces before it..'
```
This ^^^ may be fixed in #105.


```
'"/sse/priv/docroot/ping/ping.css":1.17-1.17: error: CSS: This profile has a very specific syntax for @charset: @charset followed by exactly one space, followed by the name of the encoding in quotes, followed immediately by a semicolon..'
...
```
^^^ not checked, shall check after #105 that has a related change.

```
'"/sse_cfg/priv/docroot/account/account.html":29.13-29.30: error: The value of the "for" attribute of the "label" element must be the ID of a non-hidden form control.'
'"/sse_cfg/priv/docroot/account/account.html":34.13-34.34: error: The value of the "for" attribute of the "label" element must be the ID of a non-hidden form control.'
```
^^^ unclear - not investigated.

```
'"/sse_cfg/priv/docroot/account/account.html":11.7-11.13: error: Element "right" not allowed as child of element "menu" in this context. (Suppressing further errors from this subtree.)'
...
```
^^^ seems quite fundamentals, have not checked yet.

**Comment by lucafavatellawork** [comment]
Rebased on #105.

**Comment by lucafavatellawork** [comment]
Change base so to show only relevant commits.

File: apps/sse_cfg/priv/docroot/account/account.html [hunk 1]
```diff
@@ -28,13 +28,15 @@
...
+              type="text" name="user" class="name"/>
+            </label>
           </account>
 
           <password>
-            <label for="password">Password</label>
-            <input type="password" name="password" class="password"/>
+            <label>Password<input
+              type="password" name="password" class="password"/>
+            </label>
```
Comment (lucafavatellawork) [inline apps/sse_cfg/priv/docroot/account/account.html]: This changes slightly the appearance of the form at `/account/account.html`.

File: apps/sse_cfg/priv/docroot/account/account.html [hunk 2]
```diff
@@ -28,13 +28,15 @@
...
+              type="text" name="user" class="name"/>
+            </label>
           </account>
 
           <password>
-            <label for="password">Password</label>
-            <input type="password" name="password" class="password"> </input>
+            <label>Password<input
+              type="password" name="password" class="password"/>
+            </label>
```
Comment (lucafavatellawork) [inline apps/sse_cfg/priv/docroot/account/account.html]: This changes slightly the appearance of the form at `/account/account.html`.

File: .travis.yml [hunk 3]
```diff
@@ -13,6 +13,24 @@ jobs:
...
+      python:
+        - "3.6"
+      addons:
+        apt:
+          packages:
+            - openjdk-8-jre
+      install:
+        - pip install html5validator
+      script:
+        - html5validator --root apps/ --also-check-css --ignore-re 'Element "[^"]+" not allowed as child of element "[^"]+" in this context' --ignore 'A "link" element with a "sizes" attribute must have a "rel" attribute that contains the value "icon" or the value "apple-touch-icon" or the value "apple-touch-icon-precomposed".'
```
Comment (lucafavatellawork) [inline .travis.yml]: Re:
> `--ignore-re 'Element "[^"]+" not allowed as child of element "[^"]+" in this context'`

from https://en.wikipedia.org/wiki/HTML5#Differences_between_the_two_standards
> Content or Features Unique to W3C or WHATWG Standard
> | | W3C | WHATWG |
> ...
> | Chapter Elements of HTML | | §4.13 Custom elements[[54](https://html.spec.whatwg.org/#custom-elements)] |

though https://html.spec.whatwg.org/#custom-elements reads:
> ... Although authors could always use non-standard elements in their documents, with application-specific behavior added after the fact by scripting or similar, such elements have historically been non-conforming and not very functional. ...

**Comment by lucafavatellawork** [comment]
> the same for Javascript

It is done in the CI.

> Please could you have the HTML linting occur as part of the `make lint` target, so it's part of the other linting (and dialyzing)?

Sure. As per review and chat, I will:
- [ ] Not enforce double quotes (`"`) in `@charset` in CSS.
- [ ] Move HTML/CSS/JS linting to per-application `lint` target (potentially referring to a dummy file for catering for apps without any such files).

**Comment by jacobythwaites** [comment]
I remain unsure about this. HTML and CSS are really variable and non-canonical and I don't see the need for linting them. The current use case is our web UI built into the sse_core apps' priv/docroot directories, and I think there is a strong case for getting rid of that UI entirely because it can be hosted on an external site just like any other integration.

**Comment by lucafavatellawork** [comment]
No point in addressing comments then. Closing without merging.

**Pull request closed without merging**